### DISCUSSION TITLE
KQL for search CVE in Sentinel

### DISCUSSION THREAD

**Discussion by fpsilva-source** [rod-trent/MustLearnKQL#24] [Q&A]
What tip can you pass to create a query to validate if a particular CVE has generated an incident in the environment?

For example, CVE XXXX, in my environment I have incident related to it.

**Comment by SQLtattoo** [comment]
Hi,
 well, there is no easy integration currently for this, meaning bringing the M365D TVM information in the Sentinel workspace, at least to my knowledge. I see that it is coming, though:

 (partial image from the Microsoft 365 Defender connector that is currently in preview)
![image](https://user-images.githubusercontent.com/11569518/188665840-577f0d56-417e-45d2-bc8e-acc4938c0507.png)

You can read more **[here](https://techcommunity.microsoft.com/t5/microsoft-sentinel-blog/what-s-new-closer-integration-between-microsoft-sentinel-and/ba-p/3299675)**.

Now, if you want to have this info from the Sentinel there are solutions with 3rd party solutions that you can deploy on Sentinel workspace i.e. Tenable TVM you can read a related blog post **[here](https://techcommunity.microsoft.com/t5/microsoft-sentinel-blog/how-to-integrate-vulnerability-management-in-azure-sentinel/ba-p/1635728)**.

In case you want to do it with M365D you can go to [https://security.microsoft.com](https://security.microsoft.com) and then go to "Advanced Hunting" and run the following type of query:

````
DeviceTvmSoftwareVulnerabilities 
| where CveId =~ "CVE-2022-33631"
````
and you will get something like that back:

![image](https://user-images.githubusercontent.com/11569518/188668681-44195eed-bdb2-4f50-ab8d-44da33495fa0.png)

Don't forget that this is coming soon to Sentinel, so as always patience will be rewarding :)

HTH
Vassilis

**Comment by SQLtattoo** [comment]
I see that the Tenable TVM solution is also available at the "**Content Hub**" inside the Sentinel's environment.
So you can take it for test drive from there directly :)

**Comment by fpsilva-source** [comment]
Thanks for the information, it will help a lot. I also need to associate the CVE with a certain incident. For example, in my entire incident history, I have someone associated with CVE-2022-33631. I need to find out if I have and if so how many incidents associated with a given CVE.

**Comment by CliveWatsonQC** [comment]
You can sometimes find other CVE info, but as above it depends what Tables you have and if there is any CVE to be found, a quick check would be: 

search "CVE-2022"
| where TimeGenerated > ago(7d)
| summarize count() by Type
